### DISCUSSION TITLE
post/durable-functions-di

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#22] [Announcements]
# post/durable-functions-di

Mark Heath's Development Blog

https://markheath.net/post/durable-functions-di

**Comment by renko7** [comment]
I was injecting ILogger through the constructor and was unable to see logs being outputted. However, when I injected ILogger through the method I was able to see the logs being outputted. I wanted to use constructor injection and could not figure out why I was getting this issue. Your solution of updating the hosts.json worked for me. Thank you very much for providing this resource. Do you know why the logs automatically get filtered automatically? Not a fan of that feature. Thanks.

**Reply by markheath** [reply]
that's probably a question for Microsoft. I guess they are filtered to avoid noisy logging.

**Comment by zagoskin** [comment]
I was wondering, since the microsoft example uses both static for the orchestration and the orchestrator, is it safe to call non-static activities from a static durable client that starts a static orchestration?

**Reply by markheath** [reply]
yes, there is no actual distinction in Durable Functions between a "static" and "non-static" activity. It's just how you declare them in C#, and how you inject dependencies into them.

**Comment by opc51** [comment]
You will be pleased to know that anytime I read one of your articles I hear your voice reading them at 1.3 speed :)

**Comment by opc51** [comment]
A serious question this time.

Short version
All the examples of DI into Durable Functions I have seen so far are where the DI is done into the HTTP Starter function.

I want to inject into a activity function. Is this possible? Does any reason exist why this should not be possible?

More detail
I have created a "typed HTTP Client"  (https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-7.0#typed-clients). 

I am looking to inject it into the Activity function and get it to call the StarWars API (https://swapi.dev/)

I intend to use this in an Act

**Reply by markheath** [reply]
Sure, my blog post shows an example of (constructor) injection with an ActivityTrigger.
